### PR TITLE
docs(blog): add Refine and Material UI usage docs

### PR DESCRIPTION
**Pull request by necatiozmen**
I'm writing a blog post that explains creating a refine app with Material UI and Strapi data provider.
I'll continue to add new steps to the post.

### CODE REVIEW

File: documentation/blog/2022-07-21-admin-panel-with-materialui-and-strapi.md [hunk 1]
```diff
@@ -0,0 +1,503 @@
...
+- [Refine Material UI Tutorial](https://refine.dev/docs/ui-frameworks/mui/tutorial/)
+
+
+
+##  What are the benefits of using Refine?
+
+Refine is a headless React internal tool framework that helps you develop quickly while developing both B2B and B2C applications. t speeds you up while allowing full customization, making it an ideal choice for rapid development with pro features.
+
+-   It is Open Source under the MIT license.
+-   Easy to use and it is easy to learn. There are many examples to help you get started, as well as documentation.
```
Comment (ozkalai) [inline documentation/blog/2022-07-21-admin-panel-with-materialui-and-strapi.md:47]:
```suggestion
-   It is easy to use and learn. There are many examples to help you get started, as well as documentation.
```

File: documentation/blog/2022-07-21-admin-panel-with-materialui-and-strapi.md [hunk 2]
```diff
@@ -0,0 +1,503 @@
...
+-   Is a framework that does not require you to use any UI libraries or frameworks.
+-   Supports Ant Design and Material UI for quick and easy solutions.
+-   Backend agnostic, so you can connect to any backend you want.
+-   Customizable, which means you can change it to fit your needs.
+-   Some of main features are data fetching and state management, routings, authentication, authorization, internationalization, real-time, mutation modes with optimistic and pessimistic and undoadable modes
```
Comment (ozkalai) [inline documentation/blog/2022-07-21-admin-panel-with-materialui-and-strapi.md:52]:
```suggestion
-   Some of the main features are data fetching and state management, routings, authentication, authorization, internationalization, real-time, mutation modes with optimistic and pessimistic and undoable modes
```

File: documentation/blog/2022-07-21-admin-panel-with-materialui-and-strapi.md [hunk 3]
```diff
@@ -0,0 +1,503 @@
...
+    return (
+        <List>
+            <DataGrid {...dataGridProps} columns={columns} autoHeight />
+        </List>
+    );
+};
+
+```
+
+We use Material UI components that exports from Refine.
```
Comment (ozkalai) [inline documentation/blog/2022-07-21-admin-panel-with-materialui-and-strapi.md:207]:
```suggestion
We use Material UI components exports from Refine.
```

File: documentation/blog/2022-07-21-admin-panel-with-materialui-and-strapi.md [hunk 4]
```diff
@@ -0,0 +1,503 @@
...
+
+[Get more information about components and hooks in PostList page we defined&#8594](https://refine.dev/docs/ui-frameworks/mui/tutorial/#creating-a-list-page)
+
+
+[Refer to Material UI offical docs to learn more about components&#8594](https://mui.com/x/react-data-grid/components/#main-content)
+
+
+
+
+Note you will need a few more files which help src/App.tsx to find your pages and posts. In the /pages folder, put this index.tsx file in it which allows everything in the posts folder to be used elsewhere.
```
Comment (ozkalai) [inline documentation/blog/2022-07-21-admin-panel-with-materialui-and-strapi.md:217]:
```suggestion
Note you will need a few more files that help src/App.tsx to find your pages and posts. In the /pages folder, put this index.tsx file in it which allows everything in the posts folder to be used elsewhere.
```

**Comment by salihozdemir** [comment]
Great Article! 👏🏻

File: documentation/blog/2022-07-21-admin-panel-with-materialui-and-strapi.md [hunk 5]
```diff
@@ -0,0 +1,1001 @@
+---
+title: Building a Refine CRUD app with Material UI and Strapi
```
Comment (omeraplak) [inline documentation/blog/2022-07-21-admin-panel-with-materialui-and-strapi.md:2]:
```suggestion
title: Building a CRUD app with Material UI and Strapi 
```

File: documentation/blog/2022-07-21-admin-panel-with-materialui-and-strapi.md [hunk 6]
```diff
@@ -0,0 +1,1001 @@
+---
+title: Building a Refine CRUD app with Material UI and Strapi 
+description: test
```
Comment (omeraplak) [inline documentation/blog/2022-07-21-admin-panel-with-materialui-and-strapi.md:3]:
```suggestion
description: test
```
...

File: documentation/blog/2022-07-21-admin-panel-with-materialui-and-strapi.md [hunk 7]
```diff
@@ -0,0 +1,1001 @@
...
+slug: build-admin-panel-with-material-ui-and-strapi
+authors: necati
+tags: [refine, fundamentals, react, javascript, low-code, internal-tools]
```
Comment (omeraplak) [inline documentation/blog/2022-07-21-admin-panel-with-materialui-and-strapi.md:6]:
```suggestion
tags: [refine, fundamentals, react, javascript, low-code, internal-tools, crud, material-ui, mui]
```

File: documentation/blog/2022-07-21-admin-panel-with-materialui-and-strapi.md [hunk 8]
```diff
@@ -0,0 +1,1025 @@
...
+
+import listpage from '@site/static/img/blog/2022-07-21-admin-panel-with-materialui-and-strapi/list-page.png';
+import createGif from '@site/static/img/blog/2022-07-21-admin-panel-with-materialui-and-strapi/create.gif'
+import deleteGif from '@site/static/img/blog/2022-07-21-admin-panel-with-materialui-and-strapi/delete-record.gif'
+import undoableGif from '@site/static/img/blog/2022-07-21-admin-panel-with-materialui-and-strapi/undoable-mode.gif'
+
+## Introduction
+We will build an admin panel that supports CRUD operations, has built in authentication, and a mutation mode feature using industry-standard best tools.
+
+UI design can be a complex and time-consuming process, but using a tool like Material UI can help to simplify the process and speed up the development cycle. In this tutorial we'll use the benefits of Material UI and to handle data fetching and mutations. We'll also integrate StrapiV4 data provider that Refine has built-in support.
```
Comment (omeraplak) [inline documentation/blog/2022-07-21-admin-panel-with-materialui-and-strapi.md:20]:
```suggestion
UI design can be a complex and time-consuming process, but a tool like [Material UI](https://mui.com) can help simplify the process and speed up the development cycle. In this tutorial, we'll use Material UI's benefits and handle data fetching and mutations. We'll also integrate the [Strapi](https://strapi.io/) data provider that Refine has built-in support.
```

File: documentation/blog/2022-07-21-admin-panel-with-materialui-and-strapi.md [hunk 9]
```diff
@@ -0,0 +1,1025 @@
...
+---
+
+
+import listpage from '@site/static/img/blog/2022-07-21-admin-panel-with-materialui-and-strapi/list-page.png';
+import createGif from '@site/static/img/blog/2022-07-21-admin-panel-with-materialui-and-strapi/create.gif'
+import deleteGif from '@site/static/img/blog/2022-07-21-admin-panel-with-materialui-and-strapi/delete-record.gif'
+import undoableGif from '@site/static/img/blog/2022-07-21-admin-panel-with-materialui-and-strapi/undoable-mode.gif'
+
+## Introduction
+We will build an admin panel that supports CRUD operations, has built in authentication, and a mutation mode feature using industry-standard best tools.
```
Comment (omeraplak) [inline documentation/blog/2022-07-21-admin-panel-with-materialui-and-strapi.md:18]:
```suggestion
We will build an **admin panel** that supports **CRUD** operations, has built-in **authentication**, and a mutation mode ([?](https://refine.dev/docs/guides-and-concepts/mutation-mode/))  feature using industry-standard best tools.
```

File: documentation/blog/2022-07-21-admin-panel-with-materialui-and-strapi.md [hunk 10]
```diff
@@ -0,0 +1,1025 @@
...
+
+UI design can be a complex and time-consuming process, but using a tool like Material UI can help to simplify the process and speed up the development cycle. In this tutorial we'll use the benefits of Material UI and to handle data fetching and mutations. We'll also integrate StrapiV4 data provider that Refine has built-in support.
+
+We'll walk through the process of listing, creating and deleting posts in a Refine application and make use of Refine's components and hooks to build out our functionality.
+
+Steps we'll cover includes:
+- What are the benefits of using Refine?
+- Bootstrapping the Refine App
+- Implementing Strapi-v4 data provider
```
Comment (omeraplak) [inline documentation/blog/2022-07-21-admin-panel-with-materialui-and-strapi.md:27]:
```suggestion
- Implementing Strapi data provider
```

File: documentation/blog/2022-07-21-admin-panel-with-materialui-and-strapi.md [hunk 11]
```diff
@@ -0,0 +1,1025 @@
...
+- Creating a list page
+- Handling relational data
```
Comment (omeraplak) [inline documentation/blog/2022-07-21-admin-panel-with-materialui-and-strapi.md:29]:
```suggestion
- Handling relational data with [Strapi Population support](https://docs.strapi.io/developer-docs/latest/developer-resources/database-apis-reference/rest/populating-fields.html)
```

File: documentation/blog/2022-07-21-admin-panel-with-materialui-and-strapi.md [hunk 12]
```diff
@@ -0,0 +1,1025 @@
...
+
+
+##  What are the benefits of using Refine?
+
+Refine is a headless React internal tool framework that helps you develop quickly while developing both B2B and B2C applications. It speeds you up while allowing full customization, making it an ideal choice for rapid development with pro features.
+
+-   It is Open Source under the MIT license.
+-   It is easy to use and learn. There are many examples to help you get started, as well as documentation.
+-   Is a framework that does not require you to use any UI libraries or frameworks.
+-   Supports Ant Design and Material UI for quick and easy solutions.
```
Comment (omeraplak) [inline documentation/blog/2022-07-21-admin-panel-with-materialui-and-strapi.md:51]:
```suggestion
-   Supports Ant Design and Material UI natively for quick and easy solutions. Thanks to the headless approach, you can easily integrate your own UI solution.
```

File: documentation/blog/2022-07-21-admin-panel-with-materialui-and-strapi.md [hunk 13]
```diff
@@ -0,0 +1,1025 @@
...
+
+? i18n - Internationalization:
+❯ No
+```
+
+CLI should be create a project and install the selected dependencies.
+
+### Implementing Strapi-v4 data provider
+Data providers are refine components making it possible to consume different API's and data services conveniently.
+The required Strapi-V4 data provider setups are added automatically by CLI wizard.
```
Comment (omeraplak) [inline documentation/blog/2022-07-21-admin-panel-with-materialui-and-strapi.md:94]:
```suggestion
The required Strapi data provider setups are added automatically by the CLI wizard.
```

File: documentation/blog/2022-07-21-admin-panel-with-materialui-and-strapi.md [hunk 14]
```diff
@@ -0,0 +1,1025 @@
...
+
+To consume Refine's Fake StrapiV4 data provider, we'll need change API URL in the project folder.
```
Comment (omeraplak) [inline documentation/blog/2022-07-21-admin-panel-with-materialui-and-strapi.md:96]:
```suggestion
To consume Refine's Fake Strapi API, we'll need to change the `API URL` in the project folder.
```

File: documentation/blog/2022-07-21-admin-panel-with-materialui-and-strapi.md [hunk 15]
```diff
@@ -0,0 +1,1025 @@
...
+- How to bootstrap refine app
+- Connecting Strapiv4 data provider to refine app.
+- Creating pages for CRUD operations
+- Implementing some of refine features like mutation mode and location sync.
+
+refine is an open source tool that rapidly and flexibly develops for CRUD admin panels or web apps. It is easy to get started with and doesn't require a lot of code. It has nice documentation that covered examples, guidelines, and tutorials using best practices. refine is constantly being updated with new features and improvements.
+
+[Refer to official refine page for more information &#8594](https://refine.dev/)
+  
+
```
Comment (omeraplak) [inline documentation/blog/2022-07-21-admin-panel-with-materialui-and-strapi.md:1025]: here we can dynamically add sandbox from StackBlitz

File: documentation/blog/2022-07-21-admin-panel-with-materialui-and-strapi.md [hunk 16]
```diff
@@ -0,0 +1,1025 @@
...
+export default App;
+```
+
+Now, we can get current information from URL as a query parameters. We can either use this link to share to others or define filter, pagination, and sorting parameters manually from changing URL parameters.
+
+ ## Conclusion
+
+In this article, we'll show you how to build a CRUD admin panel using refine and Material UI. This approach will allow you to quickly create an admin interface for your application with minimal coding. We'll start by setting up our project with the required dependencies. Then, we'll create our CRUD components using Material UI. Finally, we'll wire everything up and add some extra features from refine like mutation mode. 
+
```
Comment (omeraplak) [inline documentation/blog/2022-07-21-admin-panel-with-materialui-and-strapi.md:1014]:
```suggestion
In this article, we'll show you how to build a **CRUD admin panel** using refine and **Material UI**. This approach will allow you to quickly create an admin interface for your application with minimal coding. We'll start by setting up our project with the required dependencies. Then, we'll create our CRUD components using Material UI. Finally, we'll wire everything up and add some extra features from refine like mutation mode. 

```

File: examples/blog/material-ui/src/pages/posts/list.tsx [hunk 17]
```diff
@@ -0,0 +1,79 @@
...
+import { IPost } from "interfaces";
+
+export const PostList: React.FC = () => {
+    const { dataGridProps } = useDataGrid<IPost>({
+        metaData: {
+            populate: ["category"],
+        },
+    });
+
+    console.log("datagridprops", dataGridProps);
```
Comment (ozkalai) [inline examples/blog/material-ui/src/pages/posts/list.tsx:22]: console log

**Review by ozkalai — APPROVED** [review]

**Review by yildirayunlu — APPROVED** [review]

**Review by omeraplak — APPROVED** [review]

**Pull request merged**